src/chaosbag/BlessCurseManager.ttslua

```diff
@@ -248,25 +244,35 @@ function callFunctions(token, isRightClick)
     if success ~= 0 then tokenArrangerApi.layout() end
 end
 
-function getChaosBag()
-    local zone = getObjectFromGUID("83ef06")
-    if zone == nil then printToAll("Zone for chaosbag not found!", "Red") return end
+-- returns a formatted string with information about the provided token type (bless / curse)
+function formatTokenCount(type)
+    if type == nil then type = mode end
+    return "(" .. (numInPlay[type] - #tokensTaken[type]) .. "/" .. #tokensTaken[type] .. ")"
+end
+
+-- called by cards that seal bless/curse tokens
+---@param param Table This contains the type and guid of the sealed token
+function sealedToken(param)
+    table.insert(tokensTaken[param.type], param.guid)
+    broadcastCount(param.type)
+end
 
-    local items = zone.getObjects()
-    local chaosbag = nil
-    for _, v in ipairs(items) do
-        if v.getDescription() == "Chaos Bag" then
-            chaosbag = getObjectFromGUID(v.getGUID())
+-- called by cards that seal bless/curse tokens
+---@param param Table This contains the type and guid of the released token
+function releasedToken(param)
+    for i, v in ipairs(tokensTaken[param.type]) do
+        if v == param.guid then
+            table.remove(tokensTaken[param.type], i)
             break
         end
     end
-    if chaosbag == nil then printToColor("No chaos bag found", playerColor) end
-    return chaosbag
-end
-
-function getTokenCount(type)
-    if type == nil then type = mode end
-    return "(" .. (numInPlay[type] - #tokensTaken[type]) .. "/" .. #tokensTaken[type] .. ")"
+    if not updating then
+        updating = true
+        Wait.time(function()
```

Comment:
This is to collapse multiple calls during the same update, correct?  As long as it's looping from the same trigger event it will all be on one frame, so you can use Wait.frames(..., 1) to tighten up the timing here